src/patternfly/components/TreeView/tree-view.scss

```diff
@@ -0,0 +1,196 @@
+      }
+
+      .pf-c-tree-view__link {
+        --pf-c-tree-view__link--PaddingLeft: var(--pf-c-tree-view__list--m-nested-level-3__list-item__link--PaddingLeft);
+      }
+    }
+  }
+}
+// stylelint-enable
+
+.pf-c-tree-view__list-item {
+  &.pf-m-expanded {
+    > .pf-c-tree-view__link > .pf-c-tree-view__link-toggle {
+      transform: rotate(var(--pf-c-tree-view__link--m-expanded__toggle--Rotate));
+    }
+  }
+}
+
+.pf-c-tree-view__link {
+  display: flex;
+  align-items: center;
+  width: 100%;
+  padding: var(--pf-c-tree-view__link--PaddingTop) var(--pf-c-tree-view__link--PaddingRight) var(--pf-c-tree-view__link--PaddingBottom) var(--pf-c-tree-view__link--PaddingLeft);
+  color: var(--pf-c-tree-view__link--Color);
+  text-align: left;
+  border: 0;
+
+  &.pf-m-current {
+    --pf-c-tree-view__link--Color: var(--pf-c-tree-view__link--m-current--Color);
+
+    font-weight: var(--pf-c-tree-view__link--m-current--FontWeight);
+  }
+
+  &.pf-m-first {
+    font-weight: var(--pf-c-tree-view__link--m-first--FontWeight);
+  }
+
+  &:hover {
+    background-color: var(--pf-c-tree-view__link--hover--BackgroundColor);
+  }
+
+  .pf-c-check {
+    --pf-c-check__input--MarginTop: 0;
+    --pf-c-check--GridGap: 0;
+
+    padding-right: var(--pf-c-tree-view__link--c-check--PaddingRight);
+  }
+
+  .pf-c-badge {
+    margin-left: var(--pf-c-tree-view__link--c-badge--MarginLeft);
+
+    &.pf-m-read {
+      --pf-c-badge--m-read--BackgroundColor: var(--pf-c-tree-view__link--c-badge--BackgroundColor);
+    }
+  }
+}
+
+.pf-c-tree-view__link-toggle {
+  display: inline-flex;
+  align-items: center;
+  justify-content: center;
+  min-width: var(--pf-c-tree-view__link-toggle--MinWidth);
+  min-height: var(--pf-c-tree-view__link-toggle--MinHeight);
+  padding-right: var(--pf-c-tree-view__link-toggle--PaddingRight);
+  padding-left: var(--pf-c-tree-view__link-toggle--PaddingLeft);
+}
+
+.pf-c-tree-view__link-text {
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+}
+
+.pf-c-tree-view__search {
+  padding: var(--pf-c-tree-view__search--PaddingTop) var(--pf-c-tree-view__search--PaddingRight) var(--pf-c-tree-view__search--PaddingBottom) var(--pf-c-tree-view__search--PaddingLeft);
+}
+
+.pf-c-tree-view__link-icon {
+  padding-right: var(--pf-c-tree-view__link-icon--PaddingRight);
+  color: var(--pf-c-tree-view__link-icon--Color);
+}
+
+.pf-c-tree-view__link-wrapper {
+  display: grid;
+  grid-template-columns: minmax(0, 100%) auto;
+
+  &:hover {
+    background-color: var(--pf-c-tree-view__link--hover--BackgroundColor);
+
+    .pf-c-tree-view__action {
+      display: block;
+      visibility: visible;
+    }
+  }
+}
+
+.pf-c-tree-view__action {
+  display: none;
+  padding-right: var(--pf-c-tree-view__action--PaddingRight);
+  padding-left: var(--pf-c-tree-view__action--PaddingLeft);
+  margin-left: auto;
+  color: var(--pf-c-tree-view__action--Color);
+  visibility: hidden;
+  border: 0;
+
+  &.pf-m-favorite {
```

Comment:
if this should be the same as the favorite in other components, the star should be smaller and have hover/focus color changes. Also where are favorited items stored in the layout? Or is that all left up to the user?

I wonder if favoriting should be saved for another sprint, too, if it's a lot of work and not required for the initial component.

Reply:
@mcoker Agree that the action should have a hover state. However, the favorite items are not stored in a layout, because I think that it's a different scenario/pattern. The general idea for the tree view menu was to have any action item on the right (depends on a use-case). If we need a pattern for "Favorite items", then it needs more exploration, but IMHO should be possible to build a place where these items are stored from existing components. For example, add the "Favorite items" tab above the tree or drill down menu where you can see the list of items you selected.

@mcarrano What do you think? Do we have use-cases for "Selected/Favorite" items? I have a use-case for selected items in a multi-level drill-down menu ([link here](https://marvelapp.com/prototype/66b0059/screen/71366718)). But, I still think that it can be implemented from existing components.

Reply:
My concern is that we already have a specific favoriting pattern in lists like this, so it seems like it would be inconsistent if this pattern didn't match the other implemenations -- at least in the design, but also potentially in the interaction pattern

https://www.patternfly.org/v4/documentation/core/components/applauncher#favorites
https://www.patternfly.org/v4/documentation/core/components/select#favorites

Reply:
@mcoker ooohh, that makes sense. Then I think that we need to discuss the other implementation to match everything else. This feature wasn't required for the initial component. Thanks for the links :)

Reply:
@mcoker @doruskova Trying to catch up on this. I agree that processing favorites in the tree menu in the same way that we do for other menus might not work. I don't know that we have a requirement for this, so maybe we should just remove favoriting  from the examples for now and we can revisit this if or when it comes up?

Reply:
@mcarrano I agree with removing favoriting for now and we can add it later because it needs more exploration. It wasn't required for the initial component.

Reply:
Sounds good, will remove the favoriting examples and we can follow-up when its needed